File: chapter-8/8.3 动态保护/DynamicProtect/DynamicProtect/ViewController.m
//
//  ViewController.m
//  DynamicProtect
//
//  Created by monkey on 2017/9/24.
//  Copyright © 2017年 Coder. All rights reserved.
//

#import "ViewController.h"
#import "AntiJailbreak.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    if(isJailbreak()){
        NSLog(@"越狱设备");
    }else{
        NSLog(@"非越狱设备");
    }
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: chapter-6/6.4-Hook/SwiftDemo/SwiftDemo/ViewController.swift
//
//  ViewController.swift
//  SwiftDemo
//
//  Created by monkey on 2017/10/17.
//  Copyright © 2017年 Coder. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func onClick(_ sender: UIButton) {
        let result = self.CustomMethod(number: 10)
        print("origin: \(result)")
    }
    
    func CustomMethod(number: Int) -> Int{
        return number + 10
    }
}

